src/Components/PaperEditor/ProgrammeScriptContainer/index.js

```diff
@@ -205,12 +211,14 @@ const ProgrammeScriptContainer = (props) => {
     const newElements = arrayMove(elements, oldIndex, newIndex);
     setElements(newElements);
     setResetPreview(true);
+    handleSaveProgrammeScript();
   };
 
   const getInsertElementIndex = () => {
     const insertElement = elements.find((el) => {
       return el.type === 'insert';
     });
+    handleSaveProgrammeScript();
```

Comment:
I _think_ this function just returns the index position of the 'insert' marker, which doesn't get saved to the database. Instead of calling `handleSaveScript` here, I think you want to add it to line 263 - basically, wherever you see `setElements` called, because that's resetting the elements in the state of this component and those changes need to be reflected in the db.